src/contexts/events.tsx

```diff
@@ -35,13 +39,30 @@ export interface WalletConnectEvent {
 }
 
 export interface TransactionEvent {
-  event: 'transaction_confirmation' | 'kyc_completed' | 'transaction_success' | 'transaction_failure';
+  event:
+    | 'transaction_confirmation'
+    | 'kyc_started'
+    | 'kyc_completed'
+    | 'signing_requested'
+    | 'transactions_signed'
+    | 'transaction_success'
+    | 'transaction_failure';
   from_asset: string;
   to_asset: string;
   from_amount: string;
   to_amount: string;
 }
 
+export interface ProgressEvent {
+  event: 'progress';
+  phase: number;
```

Comment:
Could we also send the phase message or key? Perhaps easier to keep track than the integer alone.

Reply:
We could but do you think it would improve the tracking? 🤔 I would imagine that the index is enough, and it's also easier to visualize than keys/strings. I can add it, no problem. But I'm not sure if it would be used in practice.

Reply:
I think it only makes sense when manually exploring the events and avoid having to look for what phase does this particular integer represent. No idea how this will be used for analytics, probably in that case it doesn't make sense.